src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import duke.exception.*;
+import duke.task.Task;
+
+public class Parser {
+
+	private final TaskList tasks;
+	private boolean isExit;
+
+	/**
+	 * Adds user inputted String to list and prints the user added tasks and the current number of tasks, else
+	 * print error message for the error
+	 *
+	 * @param userInput user inputted String
+	 */
+	public void commands(String userInput) {
+		Task newTask;
+		String[] tokens = userInput.split(" ");
+		try {
+			switch (tokens[0]) {
+			case "done":
+				tasks.completeAndPrintTask(userInput);
+				return;
+			case "delete":
+				tasks.deleteAndPrintTask(userInput);
+				return;
+			case "todo":
+				newTask = tasks.makeToDoTask(userInput);
+				break;
+			case "deadline":
+				newTask = tasks.makeDeadlineTask(userInput);
+				break;
+			case "event":
+				newTask = tasks.makeEventTask(userInput);
+				break;
+			case "bye":
+				if (tokens.length > 1) {
+					throw new InvalidCommandException();
+				}
+				UI.bye();
+				this.isExit = true;
+				return;
+			case "list":
+				if (tokens.length > 1) {
+					throw new InvalidCommandException();
+				}
+				tasks.printList();
+				return;
+			case "find":
+				tasks.findTasks(userInput);
+				return;
+			default:
+				throw new InvalidCommandException();
+			}
+		} catch (InvalidCommandException | NoTaskDescriptionException | NoDateIndicatorException
+				| NoDateException | InvalidFindException err) {
```

Comment:
Instead of catching so many exceptions, I think it might be better to let all these exceptions extend from `DukeException` and just catch `DukeException`?